services/streammanager/wunderkind/wunderkindmanager.go

```diff
@@ -0,0 +1,122 @@
+package wunderkind
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/aws/aws-sdk-go/service/lambda"
+	jsoniter "github.com/json-iterator/go"
+
+	"github.com/rudderlabs/rudder-go-kit/awsutil"
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	obskit "github.com/rudderlabs/rudder-observability-kit/go/labels"
+	"github.com/rudderlabs/rudder-server/services/streammanager/common"
+)
+
+const (
+	InvocationType       = "RequestResponse"
+	WunderkindRegion     = "WUNDERKIND_REGION"
+	WunderkindIamRoleArn = "WUNDERKIND_IAM_ROLE_ARN"
+	WunderkindExternalId = "WUNDERKIND_EXTERNAL_ID"
+	WunderkindLambda     = "WUNDERKIND_LAMBDA"
+)
+
+var jsonFast = jsoniter.ConfigCompatibleWithStandardLibrary
+
+type inputData struct {
+	Payload string `json:"payload"`
+}
+
+type Producer struct {
+	conf   *config.Config
+	client lambdaClient
+	logger logger.Logger
+}
+
+type lambdaClient interface {
+	Invoke(input *lambda.InvokeInput) (*lambda.InvokeOutput, error)
+}
+
+// NewProducer creates a producer based on destination config
+func NewProducer(conf *config.Config) (*Producer, error) {
```

Comment:
Let's also pass logger as a DI.

Reply:
Why logger is need to be pass in constructor? We can initialize the logger in constructor itself per destination.

Reply:
1. It better to pass it as `DI` rather than initialising it per destination and avoid using globals using `logger.NewLogger()`.
2. `logger.NewLogger().Child("streammanager").Child("wunderkind")` Right now we are doing something like this to create a child logger which requires knowledge of the parent package which seems redundant to do for every destination.
3. Also, in tests we can pass `logger.NOP` to avoid noise of logs in tests.

Reply:
This requires changing it for every destination. Right now, for all destinations except Wunderkind, we are using global initialization for the logger. We can take this as a separate task for refactoring.

Reply:
You don't need to change it to all destination, just do the same way u did for config.